src/test/java/com/team3/central/services/OrganizerSerivceTest.java

```diff
@@ -1,5 +1,144 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.team3.central.repositories.OrganizerRepository;
+import com.team3.central.repositories.SessionTokenRepository;
+import com.team3.central.repositories.entities.ConfirmationToken;
+import com.team3.central.repositories.entities.OrganizerEntity;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+class OrganizerServiceTest {
+  private OrganizerService organizerService;
+  @Mock
+  private OrganizerRepository organizerRepository;
+  @Mock
+  private SessionTokenRepository sessionTokenRepository;
+  @Mock
+  private BCryptPasswordEncoder bCryptPasswordEncoder;
+  @Mock
+  private ConfirmationTokenService confirmationTokenService;
+  @Mock
+  private EmailService emailService;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.openMocks(this);
+    organizerService = new OrganizerService(organizerRepository, sessionTokenRepository,
+        bCryptPasswordEncoder, confirmationTokenService, emailService);
+  }
+
+  @Test
+  void testSignUpWithNewOrganizer() {
+    //given
+    String name = "John";
+    String email = "john@example.com";
+    String password = "password";
+    OrganizerEntity organizerEntity = new OrganizerEntity(name, email, password);
+    when(organizerRepository.findByEmail(email)).thenReturn(Optional.empty());
+    when(bCryptPasswordEncoder.encode(password)).thenReturn("encodedPassword");
+    when(organizerRepository.save(any(OrganizerEntity.class))).thenReturn(organizerEntity);
+
+    //when
+    ResponseEntity<OrganizerEntity> responseEntity = organizerService.signUp(name, email, password);
+
+    //then
+    assertEquals(HttpStatus.CREATED, responseEntity.getStatusCode());
+    assertEquals(organizerEntity.getId(), responseEntity.getBody().getId());
+    assertEquals(organizerEntity.getEmail(), responseEntity.getBody().getEmail());
+    assertEquals(organizerEntity.getName(), responseEntity.getBody().getName());
+    assertEquals(organizerEntity.getEvents(), responseEntity.getBody().getEvents());
+    assertFalse(organizerEntity.getIsAuthorised());
+    verify(organizerRepository, times(1)).findByEmail(email);
+    verify(bCryptPasswordEncoder, times(1)).encode(password);
+    verify(organizerRepository, times(1)).save(any(OrganizerEntity.class));
+    verify(emailService, times(1)).sendSimpleMessage(eq(email), eq("Verify Your account"),
+        anyString());
```

Comment:
we use verify mostly if we have re-try logic, change of state in class or we need to test if void function was invoked. Here it is not necessary